src/math.cpp

```diff
@@ -759,25 +790,34 @@ auto ult(const CallContext& ctx) -> Value {
     // possible to correctly get the numbers back. So no origin is needed
     return math_helper<bool>(
         ctx,
-        [](double m, double n) -> bool {
-            double num_m;
-            double fraction = std::modf(m, &num_m);
+        [](Number m, Number n) -> bool {
+            unsigned long m_int;
+            if (m.is_int()) {
+                m_int = m.try_as_int();
+            } else {
+                double num_m;
+                double fraction = std::modf(m.as_float(), &num_m);
+                if (fraction != 0.0) {
+                    throw std::runtime_error(
+                        "bad argument #1 to 'ult' (number has no integer representation)");
+                }
+                m_int = (int)num_m;
```

Comment:
This is not correct. Because you can call ult with float values (but they have to be whole numbers). That's why it says "no integer **representation**".

Reply:
ok, then i did remember it wrong and the code is fine like that